### ISSUE TITLE
Using `deadpool_redis::Pool` with `apalis::redis::RedisStorage`

### ISSUE THREAD

**Issue by zakstucke** [geofmureithi/apalis#290]
Amazing library! 

I'm currently creating `apalis::redis::RedisStorage` like so:
```rust
let storage = apalis::redis::RedisStorage::new_with_config(
    apalis::redis::connect(redis_url).await.unwrap(),
    redis_config,
);
```

But, in my current app I use `deadpool_redis` to manage connections for all other redis usage:
```rust
let pool = deadpool_redis::Config::from_url(redis_url).create_pool(Some(Runtime::Tokio1)).unwrap();
let conn = pool.get().await
```

The resulting `conn` is `deadpool_redis::Connection`, which implements `redis::aio::ConnectionLike` and can therefore be used just like a regular `redis::aio::Connection`. 

Would it be possible to support a more generic redis interface object, rather than needing to supply `redis::aio::ConnectionManager` to allow alternate storages such as deadpool to be used?
At the moment it means people such as me have 2 separate redis connection managers running, which I'm sure is negative for performance.

https://crates.io/crates/deadpool-redis

**Comment by zakstucke** [comment]
Great! so I'm guessing this would involve: 
- `RedisStorage` generic over `C: redis::aio::ConnectionLike` 
- `RedisStorage.conn: ConnectionManager` -> `RedisStorage.conn: C`
- Internal usage of `RedisStorage.conn` to use `redis::aio::ConnectionLike` interface instead

**Comment by geofmureithi** [comment]
This has been resolved in 0.6
